File: test/Systems_With_Delays/DDE_tests.jl
println("Running DDE tests")

include("BC_Model.jl")
include("CV_Model.jl")
include("RS_Model.jl")

using DataFrames, CSV

bc_et = BC_Model.Run_BC_Model()
cv_et = CV_Model.Run_CV_Model()
rs_et = RS_Model.Run_Repressilator_Model()

println("DDE tests completed successfully")

data = DataFrame(model=["BC", "CV", "RS"], execution_time=[bc_et, cv_et, rs_et])
CSV.write("dde_elapsed_times_jl.csv", data)

File: test/Systems_With_Delays/DDE_tests.cpp
#include "BC_Model.hpp"
#include "CV_Model.hpp"
#include "RS_Model.hpp"

#include <fstream>

int main() 
{
    double bc_et = BC_Model();
    double cv_et = CV_Model();
    double rs_et = RS_Model();

    // save the execution times to time_performance.csv
    std::ofstream file("dde_elapsed_times_cpp.csv");
    file << "model,execution_time\n";
    file << "BC," << bc_et << "\n";
    file << "CV," << cv_et << "\n";
    file << "RS," << rs_et << "\n";
    file.close();


    return 0;
}